### DISCUSSION TITLE
searchParams is empty in route handler

### DISCUSSION THREAD

**Discussion by baba43** [vercel/next.js#52780] [Help] [answered]
### Summary

The following code is working locally but searchParams is empty when running on vercel.

Any ideas why / how to solve this?

```
import { NextRequest } from "next/server";

export async function GET(req: NextRequest) {
  const param = req.nextUrl.searchParams.get("test");
  console.log(param ); // null on vercel
  //
}
```

**Comment by icyJoseph** [comment]


**Reply by baba43** [reply]
Thank you. That did not help, but I found the reason and solution.

**Reply by icyJoseph** [reply]
what was it? :)

**Reply by baba43** [reply]
I just wrote an answer. Thank you very much for the help, again.

**Comment by baba43** [comment, marked as answer]
This issue happened for me because I was using the platforms middleware:
https://github.com/vercel/platforms/blob/main/middleware.ts

The function has to be modified so that it also passes url parameters:

```
...
const params = url.searchParams;
return NextResponse.rewrite(
    new URL(`/${hostname}${path}${params ? `?${params}` : ""}`, req.url)
);
```
